### PR TITLE
Fixes big oversight in benefit calculations

### PR DESCRIPTION
**Pull request by knod**
though it doesn't seem to have affected the current cliff example.

Until now SNAP calculations have not been affected by Section 8 benefits. Should be fixed now. Bit embarrassed about this and definitely want another eye on it to make sure it's getting done right this time.

### CODE REVIEW

File: src/forms/output/BenefitsTable.js [hunk 1]
```diff
@@ -27,22 +27,25 @@ const getSignSymbol = function (num) {
...
+    sec8BenefitFuture      = Math.round(getSection8Benefit(clone, 'future'));
+    // Mutate clone for correct SNAP values
+    clone.future.rentShare = sec8BenefitFuture;
   }
 
-  if (curr.hasSection8) {
-    sec8BenefitCurrent = Math.round(getSection8Benefit(client, 'current'));
-    sec8BenefitFuture  = Math.round(getSection8Benefit(client, 'future'));
+  if (curr.hasSnap) {
+    console.log(clone.future.rentShare);
```
Comment (wacii) [inline src/forms/output/BenefitsTable.js]: Errant `console.log`.

File: src/utils/charts/getChartData.js [hunk 2]
```diff
@@ -36,32 +75,85 @@ getData.section8 = function (xRange, client, multiplier) {
...
+
+  // Don't loop if there's nothing to do
+  if (benefitDatasets.length === 0) {
+    return benefitDatasets;
+  }
+
+  // Otherwise, loop over incomes
+  for (var incomei = 0; incomei < xRange.length; incomei++) {
+    var income = xRange[ incomei ];
+    clone.future.earned = income / multiplier;
```
[the remark this answers is not in the document]
Comment (knod) [reply src/utils/charts/getChartData.js]: We're talking it over in real-time since I think we're not on the same page. Regardless, I'm not sure I've expressed this well here, so it obviously needs to be clarified somehow.

File: src/utils/charts/getChartData.js [hunk 3]
```diff
@@ -36,32 +75,85 @@ getData.section8 = function (xRange, client, multiplier) {
...
 };  // End getData.section8()
 
+/**
+ * @param {array} xRange All income values to be included.
+ */
+const insertBenefitData = function (xRange, clone, multiplier, benefitDatasets) {
+
+  // Don't loop if there's nothing to do
+  if (benefitDatasets.length === 0) {
+    return benefitDatasets;
```
[the remark this answers is not in the document]
Comment (knod) [reply src/utils/charts/getChartData.js]: Fair point, a return value can be confusing in a function that's meant to mutate data. And it should also be documented. I wish there were an `@mutates` in jsdoc or something. That'd be neat.

File: src/utils/charts/getChartData.js [hunk 4]
```diff
@@ -36,32 +75,85 @@ getData.section8 = function (xRange, client, multiplier) {
...
+    var income = xRange[ incomei ];
+    clone.future.earned = income / multiplier;
+
+    // Datasets need to be in correct order for benefits to
+    // affect each other corrrectly.
+    for (let benefiti = 0; benefiti < benefitDatasets.length; benefiti++) {
+
+      let dataset = benefitDatasets[ benefiti ],
+          name    = dataset.benefitName,
+          funcs   = getBenefit[ name ];
```
Comment (knod) [inline src/utils/charts/getChartData.js]: @wacii wrote this, and it's about to be oudated by an unrelated change:
This inner loop and `getBenefit` are too strongly coupled. I would rather see `getBenefit[name]` resolve to a function that does both the subsidy calculation and the update of the client (when necessary).

**Comment by knod** [comment]
Or maybe that commit didn't fix everything...

File: src/utils/charts/getChartData.js (hunk 4 above)
Comment (knod) [reply src/utils/charts/getChartData.js]: I like this idea, and it's one I'll work on, but maybe on a different PR? I'd like to get this fix through, even if it's just a temporary one, so that we have a tool that's working.

**Comment by knod** [comment]
@todo on this one.
